src/ocean/text/arguments/DateIntervalArgs.d

```diff
@@ -0,0 +1,429 @@
+/*******************************************************************************
+
+    Handler for date interval CLI arguments
+
+    copyright: Copyright (c) 2018 dunnhumby Germany GmbH. All rights reserved
+
+*******************************************************************************/
+
+module ocean.text.arguments.DateIntervalArgs;
+
+import core.stdc.time;
+
+import ocean.util.app.model.IApplication;
+import ocean.text.Arguments;
+import ocean.text.convert.DateTime;
+import ocean.text.convert.Integer;
+import ocean.transition;
+
+version (UnitTest)
+{
+    import Test = ocean.core.Test;
+}
+
+/// Class that handles the begin-date and end-date CLI arguments
+public class DateIntervalArgs
+{
+    /// The begin and end of the run, if given on the command line
+    private long cli_begin;
+
+    /// ditto
+    private long cli_end;
+
+    /// The interval that should be used when dates are not set by the user
+    private long default_interval;
+
+    /**************************************************************************
+
+        Constructor
+
+        Params:
+            default_interval = the default interval iin seconds when the
+                arguments are missing.
+
+    **************************************************************************/
+
+    public this ( long default_interval )
+    {
+        this.default_interval = default_interval;
+    }
+
+    /***************************************************************************
+
+        Get the parsed begin-date value
+
+        Returns:
+            The interval begin as a timestamp
+
+    ***************************************************************************/
+
+    public long begin ( )
+    {
+        if ( this.cli_begin == 0 && this.cli_end == 0 )
+        {
+            return time(null) - this.default_interval;
+        }
+
+        if ( this.cli_begin == 0 )
+        {
+            return time(null);
+        }
+
+        return this.cli_begin;
+    }
+
+    /***************************************************************************
+
+        Get the parsed end-date value
+
+        Returns:
+            The interval end as a timestamp
+
+    ***************************************************************************/
+
+    public long end ( )
+    {
+        if ( this.cli_end == 0 )
+        {
+            return time(null);
```

Comment:
Same goes for this, I believe it should be based on when processArgs was called, otherwise it's an ever changing value.